pkg/vault/envconsul/webhook.go

```diff
@@ -0,0 +1,177 @@
+		vaultConfigKey:   vaultConfigKey,
+	}
+
+	for _, opt := range opts {
+		opt(&handler)
+	}
+
+	return builder.NewWebhookBuilder().
+		Name(name).
+		Mutating().
+		Operations(admissionregistrationv1beta1.Create).
+		ForType(&corev1.Pod{}).
+		Handlers(handler).
+		WithManager(mgr).
+		Build()
+}
+
+type Injector struct {
+	logger           kitlog.Logger
+	decoder          types.Decoder
+	client           client.Client
+	webhookConfigKey client.ObjectKey
+	vaultConfigKey   client.ObjectKey
+}
+
+func (i *Injector) Handle(ctx context.Context, req types.Request) types.Response {
+	logger := kitlog.With(i.logger, "uuid", string(req.AdmissionRequest.UID))
+	logger.Log("event", "request.start")
+	defer func(start time.Time) {
+		logger.Log("event", "request.end", "duration", time.Now().Sub(start).Seconds())
+	}(time.Now())
+
+	pod := &corev1.Pod{}
+	if err := i.decoder.Decode(req, pod); err != nil {
+		admission.ErrorResponse(http.StatusBadRequest, err)
+	}
+
+	podLogger := kitlog.With(logger, "pod_namespace", pod.Namespace, "pod_name", pod.Name)
+
+	config, ok := pod.Annotations[fmt.Sprintf("%s/configs", name)]
+	if !ok {
+		podLogger.Log("msg", "skipping: no configuration annotation")
+		return admission.PatchResponse(pod, pod)
+	}
+
+	vaultConfig := &corev1.ConfigMap{}
+	i.client.Get(ctx, i.vaultConfigKey, vaultConfig)
+	vaultAddress, _ := vaultConfig.Data["vault_address"]
+	vaultAuthMountPath, _ := vaultConfig.Data["auth_mount_path"]
+	vaultAuthRole, _ := vaultConfig.Data["auth_role"] // TODO: Add to the config file
+
+	webhookConfig := &corev1.ConfigMap{}
+	i.client.Get(ctx, i.webhookConfigKey, webhookConfig)
+	image, _ := webhookConfig.Data["image"]
+	installPath, _ := webhookConfig.Data["install_path"]
+
+	containerConfigPaths := getContainerConfigPaths(config)
+	copy := pod.DeepCopy()
+
+	for k, container := range copy.Spec.Containers {
+		if configFile, ok := containerConfigPaths[container.Name]; ok {
+			c := &copy.Spec.Containers[k]
+			originalCommand := strings.Join(append(c.Command, c.Args...), " ")
+			c.Command = []string{
+				path.Join(installPath, "theatre-envconsul"),
+				"exec",
+				fmt.Sprintf("--vault-address=%s", vaultAddress),
+			}
+			if installPath != "" {
+				c.Command = append(c.Command, fmt.Sprintf("--install-path=%s", installPath))
+			}
+			if vaultAuthMountPath != "" {
+				c.Command = append(c.Command, fmt.Sprintf("--auth-backend-mount-path=%s", vaultAuthMountPath))
+			}
+			if vaultAuthRole != "" {
+				c.Command = append(c.Command, fmt.Sprintf("--auth-backend-role=%s", vaultAuthRole))
+			}
+			if configFile != "" {
+				podLogger.Log("msg", fmt.Sprintf("mutating %s with config file: %s", container.Name, configFile))
+				c.Command = append(c.Command, fmt.Sprintf("--config-file=%s", configFile))
+			} else {
+				podLogger.Log("msg", fmt.Sprintf("mutating %s without config file", container.Name))
+			}
+			c.Command = append(c.Command, fmt.Sprintf("--command=%s", originalCommand))
```

Comment:
Are we double-appending here?

Would recommend only reassigning `c.Command` in one place.

Reply:
Actually this is fine, ignore!